pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BaseTableUpsertMetadataManager.java

```diff
@@ -61,11 +62,12 @@ public void init(TableConfig tableConfig, Schema schema, TableDataManager tableD
     PartialUpsertHandler partialUpsertHandler = null;
     if (upsertConfig.getMode() == UpsertConfig.Mode.PARTIAL) {
       Map<String, UpsertConfig.Strategy> partialUpsertStrategies = upsertConfig.getPartialUpsertStrategies();
-      Preconditions.checkArgument(partialUpsertStrategies != null,
+      String rowMergerCustomImplementation = upsertConfig.getPartialUpsertMergerClass();
+      Preconditions.checkArgument(
+          StringUtils.isNotBlank(rowMergerCustomImplementation) || partialUpsertStrategies != null,
           "Partial-upsert strategies must be configured for partial-upsert enabled table: %s", _tableNameWithType);
       partialUpsertHandler =
-          new PartialUpsertHandler(schema, partialUpsertStrategies, upsertConfig.getDefaultPartialUpsertStrategy(),
-              comparisonColumns);
+          new PartialUpsertHandler(schema, comparisonColumns, upsertConfig);
```

Comment:
(format) Please auto-reformat all the changes with [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#set-up-ide)